pkgs/build-support/fetchnextcloudapp/default.nix

```diff
@@ -4,18 +4,23 @@
 , sha256 ? ""
 , appName ? null
 , appVersion ? null
-, license
+, license ? null
+, licenses ? [ ]
 , patches ? [ ]
 , description ? null
 , homepage ? null
 , unpack ? false # whether to use fetchzip rather than fetchurl
 }:
+let
+  licenses' = lib.optional (license != null) license ++ licenses;
+  licenses'' = assert lib.assertMsg (lib.length licenses' > 0) "License(s) for ${url} must be specified"; licenses';
```

Comment:
I don't understand Maximilian -- that's what this code does, as I read it. In fact, if a license isn't specified, eval will fail, which is quite loud.

Reply:
The assertion will be thrown if `license` is `null` __and__ `licenses` is an empty list (which is a good idea).

However, I think we want a warning if `license` is `null`, even if `licenses` is non-empty since the `license` option is what we're deprecating.

So basically `licenses'` vs `licenses`.

Or am I misreading it? oO

Reply:
I think so! The point of the PR is that you're able to specify a license through two doors:

1. A singular license with the `license` argument
2. One or more licenses with the `licenses` argument

This PR unifies these two doors into one through treating the singular license as the first entry in `licenses`, if it exists.

The invariant which is maintained is that every package has a license.

I see no need to deprecate the `license` argument -- it's fine as long as there's a meaningful merge strategy, which @SuperSandro2000 has provided.

Reply:
Isn't it a little weird to have two ways of achieving the same thing? Especially if the `license` just gets prepended to the list of other licenses?

While I do think we should leave it that way for now, I think it makes sense to actually deprecate the `license` arg.

Reply:
It's a little odd, but not too bad, and a definite step up from the previous state of the world where it was impossible to specify more than one license.

I expect the vast majority of cases to specify through the license door -- there's just one license, and it's this one -- and a few specific cases where there's multiple licenses to use the `licenses` door, since they need to do so in order to mark their terms.

In any case, blocking the PR on forcing the majority of single cases where they are already perfectly in compliance to migrate to a plural form doesn't seem worth it to me at all. I think we ought to merge 🙂

Reply:
So the thing is, `meta.license` is poorly named because it also allows a list of licenses. If people wrongly assume this is the case here, you'll end up with a malformed `meta` section.

> I expect the vast majority of cases to specify through the license door -- there's just one license

Also I don't see why you shouldn't use a list for a single element if having multiple values is a valid use-case.

I don't think we should leave APIs in a weird state, especially if there's a trivial way forward. Even though it's one of the lesser used builders, I think we already have enough weird things?